kong/plugins/httplog/log.lua

```diff
@@ -0,0 +1,19 @@
+local cjson = require "cjson"
+local http_client = require "kong.tools.http_client"
+
+local APPLICATION_JSON_TYPE = "application/json"
+
+local _M = {}
+
+local function log(premature, conf, message)
+  local response, status, headers = http_client.post(conf.logging_url, cjson.encode(message), { ["content-type"] = APPLICATION_JSON_TYPE })
+end
+
+function _M.execute(conf)
+  local ok, err = ngx.timer.at(0, log, conf, ngx.ctx.log_message)
```

Comment:
Please add `timer.at` to the nginx_stub to allow unit tests to mock it if needed.